content/en/docs/architecture/ci-operator-internals/observer-pods.md

```diff
@@ -0,0 +1,217 @@
+
+### Disable a predefined observer
+The `observers` stanza found inside `steps` is made up in this way:
+{{< highlight yaml >}}
+observers:
+  # List of observers that will be included in the test execution
+  enable:
+  - observer-1
+  - observer-2
+  # List of observers that *won't* be included in the test execution
+  disable:
+  - observer-3
+{{< / highlight >}}
+
+The `disable` stanza let the user selectively decide what observers not to include.
+Assuming the following configuration:
+{{< highlight yaml >}}
+tests:
+- as: e2e
+  steps:
+    test:
+    - as: first-step
+      commands: echo 'this is a test'
+      ...
+      observers:
+      - observer-3
+    observers:
+      enable:
+      - observer-1
+      - observer-2
+      disable:
+      - observer-3
+{{< / highlight >}}
+The final set of observers will be: `{ observer-1, observer-2 }`.
+
+### From scratch on Step Registry
+Writing an observer as a step inside the registry looks almost like a regular test.
+
+Observer's definition file must have `-observer.yaml` suffix. This is how the configuration looks like:
+{{< highlight yaml >}}
+observer:
+  name: observer
+  # Only one of the following two should be specified
+  from: os # This image will be used to run the observer
+  # Literal image stream tag that will be used to run the observer
+  from_image:
+    ...
+  # This file has to be inside the same observer folder
+  commands: observer-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 10Mi
+  documentation: |-
+    A simple observer
+{{< / highlight >}}
+
+As it is for regular tests, observer's payload file must have `-commands` suffix (file extension has not
+been taken into account this time): 
+{{< highlight bash >}}
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+echo "I'm going to silently observe your test"
+{{< / highlight >}}
+
+The final folder content wuold be, in this case:
+- `observer-observer.yaml`
+- `observer-commands.sh`
+
+## Execution Model
+Observers are started before any test from the `pre:` chain does and they keep staying alive till
+the last step from the `test:` chain completes.  
+
+When an observer fails, its failure will be reported inside logs and jUnit files but the test
+will keep going without being affected at all.  
+
+When the last test from `test:` chain finishes, a `SIGTERM` is sent to every observer and its related pod is deleted. After the test completes, no 'observer pods' pods will be found inside `ci-op-xxxx` namespace.
+
+Artifacts are being collected as they were regular tests.
+
+It's a good habit writing observer payload following this approach:
+{{< highlight bash >}}
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+function cleanup() {
+    echo 'this will be executed last'
+}
+
+trap cleanup EXIT
+
+# Do something here
+# An observer could be killed in the middle of its execution!
```

Comment:
killed or sigterm is sent?

Reply:
Kubernetes try to send a `SIGTERM` signal firstly and then, after a grace period of n seconds, it `SIGKILL`s the container.